feature_engine/prediction/target_mean_classifier.py

```diff
@@ -0,0 +1,192 @@
+
+    pipeline_:
+        A Scikit-learn Pipeline with a dicretiser and/or encoder. Used to determine the
+        mean target value per category or bin, per variable.
+
+    n_features_in_:
+        The number of features in the train set used in fit.
+
+    Methods
+    -------
+    fit:
+        Learn the mean target value per category or per bin, for each variable.
+
+    predict:
+        Predict class labels for samples in X.
+
+    predict_log_proba:
+        Predict logarithm of probability estimates.
+
+    predict_proba:
+        Proxy for Probability estimates based of the average of the target mean value
+         across variables.
+
+    score:
+        Return the mean accuracy on the given test data and labels.
+
+    See Also
+    --------
+    feature_engine.encoding.MeanEncoder
+    feature_engine.discretisation.EqualWidthDiscretiser
+    feature_engine.discretisation.EqualFrequencyDiscretiser
+
+    References
+    ----------
+    Adapted from:
+
+    .. [1] Miller, et al. "Predicting customer behaviour: The University of Melbourne’s
+        KDD Cup report". JMLR Workshop and Conference Proceeding. KDD 2009
+        http://proceedings.mlr.press/v7/miller09/miller09.pdf
+    """
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+        Learn the mean target value per category or bin.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training input samples.
+
+        y : pandas series of shape = [n_samples,]
+            The target variable.
+        """
+        # check that y is binary
+        self.classes_ = list(y.unique())
+
+        if len(self.classes_) > 2:
+            raise ValueError(
+                "This encoder is designed for binary classification only. The target "
+                "has more than 2 unique values."
+            )
+
+        # if target has values other than 0 and 1, we need to remap the values,
+        # to be able to compute meaningful averages.
+        if any(x for x in y.unique() if x not in [0, 1]):
+            y = np.where(y == y.unique()[0], 0, 1)
+
+        super().fit(X, y)
+
+        return self
+
+    def predict_proba(self, X: pd.DataFrame) -> pd.Series:
+        """
+        Proxy for probability estimates based of the average of the target mean value
+        across variables.
+
+        The returned estimates for all classes are ordered by the label of classes.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The input samples.
+
+        Return
+        -------
+        T: array-like of shape (n_samples, n_classes)
+            Returns the probability of the sample for each class in the model, where
+            classes are ordered as they are in self.classes_.
+        """
+        prob = self._predict(X)
+        if not isinstance(prob, np.ndarray) or prob.shape[1] != 2:
```

Comment:
why this check? what is it ensuring? 

Also, is the output of _predict() not a 1-D array, always?